testing/unittests/jlibtests.cpp

```diff
@@ -3126,46 +3125,33 @@ class JlibCompressionTestsStress : public CppUnit::TestFixture
                 }
             }
 
-            DBGLOG("Algorithm || Compression Time (ms) || Decompression Time (ms) || Compression Ratio");
+            DBGLOG("Algorithm(options)  || Comp(ms) || Deco(ms) || 200MB/s (w,r)   || 1GB/s (w,r)     || 5GB/s (w,r)     || Ratio [cLen]");
+            DBGLOG("                    ||          ||          || 2Gb/s           || 10Gb/s          || 50Gb/s          ||");
 
+            unsigned time200MBs = transferTimeMs(sz, 200000000);
+            unsigned time1GBs = transferTimeMs(sz, 1000000000);
+            unsigned time5GBs = transferTimeMs(sz, 5000000000);
+            DBGLOG("%19s || %8u || %8u || %4u(%4u,%4u) || %4u(%4u,%4u) || %4u(%4u,%4u) || %5.2f [%u]", "uncompressed", 0, 0,
+                time200MBs, time200MBs, time200MBs, time1GBs, time1GBs, time1GBs, time5GBs, time5GBs, time5GBs, 1.0, sz);
             ForEach(*iter)
             {
-                compressed.clear();
                 ICompressHandler &handler = iter->query();
+                const char * type = handler.queryType();
                 //Ignore unusual compressors with no expanders...
-                if (strieq(handler.queryType(), "randrow"))
+                if (strieq(type, "randrow"))
                     continue;
-                Owned<ICompressor> compressor = handler.getCompressor(streq("AES", handler.queryType()) ? aesKey: nullptr);
-
-                CCycleTimer timer;
-                compressor->open(compressed, sz);
-                compressor->startblock();
-                const byte *ptr = src.bytes();
-                const byte *ptrEnd = ptr + src.length();
-                while (ptr != ptrEnd)
+                const char * options = streq("AES", handler.queryType()) ? aesKey: "";
+                if (streq(type, "LZ4HC"))
                 {
-                    compressor->write(ptr, rowSz);
-                    ptr += rowSz;
+                    testCompressor(handler, "hclevel=3", rowSz, src.length(), src.bytes(), false);
+                    testCompressor(handler, "hclevel=4", rowSz, src.length(), src.bytes(), false);
+                    testCompressor(handler, "hclevel=5", rowSz, src.length(), src.bytes(), false);
+                    testCompressor(handler, "hclevel=6", rowSz, src.length(), src.bytes(), false);
+                    testCompressor(handler, "hclevel=8", rowSz, src.length(), src.bytes(), false);
+                    testCompressor(handler, "hclevel=10", rowSz, src.length(), src.bytes(), false);
+                    testCompressor(handler, "hclevel=8", rowSz, src.length(), src.bytes(), true);
```

Comment:
In the output table it looks like this is a duplicate test of LZ4HC(hclevel=8) ..
Results are very similar, perhaps not worth including.

Reply:
Pushed and squashed that change.  I'll edit above to show new output.  Reducing the number of virtual calls helps a little, but avoiding the internal memcpy saves about 20% for LZ4.